app/cli/plugins/shared.go

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugins
+
+import (
+	"encoding/gob"
+
+	"github.com/hashicorp/go-plugin"
+)
+
+func init() {
+	// Register types that will be sent over RPC
+	gob.Register(map[string]any{})
```

Comment:
why do you need to register all these types? 

It's interesting that you decided to use gob instead of protobuf, which is what we use in our [other plugin implementation](https://github.com/chainloop-dev/chainloop/blob/main/app/controlplane/plugins/sdk/v1/plugin/api/fanout.proto), out of curiosity, why?

Did you get this from any example/documentation from the plugin? Is this their recommended way? I am out of touch with the library.

Reply:
From my perspective it was easier, just a matter of preference, no hard feelings I would say and I followed one of the examples. If you think we should aim for unification in the plugins across all plugins let me know.

Reply:
> why do you need to register all these types?

I simplified it - it was a leftover from the changes that I did when developing.